paasta_tools/docker_wrapper.py

```diff
@@ -71,31 +76,65 @@ def generate_hostname(fqdn, mesos_task_id):
     return hostname
 
 
-def add_hostname(args, hostname):
-    # Add --hostname argument immediately after 'run' command if it exists
+def add_argument(args, argument):
+    # Add an argument immediately after 'run' command if it exists
     args = list(args)
 
     try:
         run_index = args.index('run')
     except ValueError:
         pass
     else:
-        args.insert(run_index + 1, '--hostname=' + hostname)
+        args.insert(run_index + 1, argument)
 
     return args
 
 
+def get_core_list(cpuid):
+    core=0
+    core_list=[]
+
+    try:
+        with open('/proc/cpuinfo', 'r') as f:
+            for line in f:
+                m = re.match('physical\sid.*(\d)', line)
+                if m:
+                    if int(m.group(1)) == cpuid:
+                        core_list.append(core)
+                    core +=1
+    except:
+        pass
+
+    return core_list
+
+
+def is_numa_enabled():
+    return os.path.exists('/proc/1/numa_maps')
+
+
 def main(argv=None):
     argv = argv if argv is not None else sys.argv[1:]
 
     env_args = parse_env_args(argv)
-    fqdn = socket.getfqdn()
 
     # Marathon sets MESOS_TASK_ID whereas Chronos sets mesos_task_id
     mesos_task_id = env_args.get('MESOS_TASK_ID') or env_args.get('mesos_task_id')
 
+    # Invalid the variable if it has a bogus value
+    numa_cpuid=env_args.get('NUMA_CPU_AFFINITY')
+    try:
+        int(numa_cpuid)
+    except:
+        numa_cpuid=None
+
+    if numa_cpuid and is_numa_enabled():
+        core_list=get_core_list(int(numa_cpuid))
+        if len(core_list) > 0:
+            argv = add_argument(argv, '--cpuset-cpus=' + ','.join(str(c) for c in core_list))
```

Comment:
That decision should be made by user/scheduler. Knowing the pool of machine that we have, we should not enable this option for services that require more core than what's on a single CPU on our smallest instance type. (This value would be 8 for most of common EC2 instances).
The good news is, even if we set this by mistake : it won't break anything and Docker will enforce the limit anyway so some thread will have to actively share a core.